cdap-docs/tools/docs-search-deprecated.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+#  Copyright © 2016 Cask Data, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#
+# Reads in the Javadocs page 
+# (at http://docs.cask.co/cdap/<release-version>/en/reference-manual/javadocs/deprecated-list.html);
+# extracts all of the listed deprecated items, and then searches in the Java, text, and
+# ReST files of the examples and the documentation for any usage. 
+#
+# It may generate significant false positives as it does not perform a search with any context.
+#
+# Requires Python 2.7 and Beautiful Soup.
+
+try:
+    from bs4 import BeautifulSoup
+    from optparse import OptionParser
+    import os
+    import sys
+    import urllib2
+except Exception, e:
+    print "\nAre you running under Python 2.7? It's required.\n"
+    raise
```

Comment:
Looks like it depends on the version of 2.6 that is invoked; some cause an error message to be raised instead of the stack trace and exception. Fixed.